dataprofiler/tests/space_time_analysis/structured_space_time_analysis.py

```diff
@@ -112,67 +115,102 @@ def dp_time_analysis(
-        except ValueError:
-            pass  # empty profile merge if 0 data
-        merge_time = time.time() - start_time
-
-        # get times for each profile in the columns
-        for profile in profiler.profile:
-            compiler_times = defaultdict(list)
-
-            for compiler_name in profile.profiles:
-                compiler = profile.profiles[compiler_name]
-                inspector_times = dict()
-                for inspector_name in compiler._profiles:
-                    inspector = compiler._profiles[inspector_name]
-                    inspector_times[inspector_name] = inspector.times
-                compiler_times[compiler_name] = inspector_times
-            column_profile_time = {
-                "name": profile.name,
-                "sample_size": sample_size,
-                "total_time": total_time,
-                "column": compiler_times,
-                "merge": merge_time,
-                "percent_to_nan": percent_to_nan,
-                "allow_subsampling": allow_subsampling,
-                "is_data_labeler": options.structured_options.data_labeler.is_enabled,
-                "is_multiprocessing": options.structured_options.multiprocess.is_enabled,
-            }
-            profile_times += [column_profile_time]
-
-        # add time for for Top-level
-        if sample_size:
-            profile_times += [
-                {
-                    "name": "StructuredProfiler",
+        if time_analysis:
+            # time profiling
+            start_time = time.time()
+            if allow_subsampling:
+                profiler = dp.Profiler(df, options=options)
+            else:
+                profiler = dp.Profiler(df, samples_per_update=len(df), options=options)
+            total_time = time.time() - start_time
+
+            # get overall time for merging profiles
+            start_time = time.time()
+            try:
+                merged_profile = profiler + profiler
+            except ValueError:
+                pass  # empty profile merge if 0 data
+            merge_time = time.time() - start_time
+
+            # get times for each profile in the columns
+            for profile in profiler.profile:
+                compiler_times = defaultdict(list)
+
+                for compiler_name in profile.profiles:
+                    compiler = profile.profiles[compiler_name]
+                    inspector_times = dict()
+                    for inspector_name in compiler._profiles:
+                        inspector = compiler._profiles[inspector_name]
+                        inspector_times[inspector_name] = inspector.times
+                    compiler_times[compiler_name] = inspector_times
+                column_profile_time = {
+                    "name": profile.name,
                     "sample_size": sample_size,
                     "total_time": total_time,
-                    "column": profiler.times,
+                    "column": compiler_times,
```

Comment:
this is just the ui confusing +/- s right? it seems it is the same code just shifted.

Reply:
Yeah I didnt change any of the json output so this should not be an actual change just a shift